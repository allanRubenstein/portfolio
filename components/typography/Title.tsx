import styled from 'styled-components';
import { BREAKPOINTS } from '../../util/constants';
import { TitleTypographyProps } from './types';

export const Title = styled.span<TitleTypographyProps>`
  line-height: 0.9;
  font-size: ${(props) => props.$fontSize}rem;

  font-family: 'Roboto Condensed', sans-serif;
  font-weight: ${(props) => (props.$isBold ? 700 : 400)};
  text-shadow: ${(props) =>
    props.$hasTextShadow
      ? `${props.$fontSize / 5}px ${
          props.$fontSize / 2.5
        }px 0px rgba(0, 0, 0, .8)`
      : undefined};
  color: ${(props) =>
    `${props.$fontColor ? `var(${props.$fontColor})` : 'inherit'}`};

  /* text-transform: uppercase; */
  text-transform: ${(props) => `${props.$textTransform || 'uppercase'}`};
  /* TODO: come up with breakpoints */
  @media (min-width: ${BREAKPOINTS.medium}px) {
    ${(props) =>
      props.$desktopFontSize ? `font-size: ${props.$desktopFontSize}rem` : ''}
  }
`;
