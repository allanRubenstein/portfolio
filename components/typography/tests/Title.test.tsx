import React from 'react';
import { Title } from '../Title';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

test('button snapshot', () => {
  // default
  const { container, rerender } = render(<Title $fontSize={5}>test</Title>);
  expect(container).toMatchSnapshot();

  // desktop font size
  rerender(
    <Title $fontSize={5} $desktopFontSize={10}>
      test
    </Title>,
  );
  expect(container).toMatchSnapshot();

  // different color
  rerender(
    <Title $fontSize={5} $color="--red">
      test
    </Title>,
  );
  expect(container).toMatchSnapshot();

  // text shadow
  rerender(
    <Title $fontSize={5} $color="--red" $hasTextShadow={true}>
      test
    </Title>,
  );
  expect(container).toMatchSnapshot();
});

test('a11y', async () => {
  const { container } = render(<Title $fontSize={5}>test</Title>);
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});
