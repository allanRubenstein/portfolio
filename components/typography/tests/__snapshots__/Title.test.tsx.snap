// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button snapshot 1`] = `
.c0 {
  line-height: 0.9;
  font-size: 5rem;
  font-family: 'Roboto Condensed',sans-serif;
  font-weight: 400;
  color: inherit;
  text-transform: uppercase;
}

@media (min-width:768px) {

}

<div>
  <span
    class="c0"
  >
    test
  </span>
</div>
`;

exports[`button snapshot 2`] = `
.c0 {
  line-height: 0.9;
  font-size: 5rem;
  font-family: 'Roboto Condensed',sans-serif;
  font-weight: 400;
  color: inherit;
  text-transform: uppercase;
}

@media (min-width:768px) {

}

@media (min-width:768px) {
  .c0 {
    font-size: 10rem;
  }
}

<div>
  <span
    class="c0"
  >
    test
  </span>
</div>
`;

exports[`button snapshot 3`] = `
.c0 {
  line-height: 0.9;
  font-size: 5rem;
  font-family: 'Roboto Condensed',sans-serif;
  font-weight: 400;
  color: var(--red);
  text-transform: uppercase;
}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

<div>
  <span
    class="c0"
  >
    test
  </span>
</div>
`;

exports[`button snapshot 4`] = `
.c0 {
  line-height: 0.9;
  font-size: 5rem;
  font-family: 'Roboto Condensed',sans-serif;
  font-weight: 400;
  text-shadow: 1px 2px 0px rgba(0,0,0,.8);
  color: var(--red);
  text-transform: uppercase;
}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

<div>
  <span
    class="c0"
  >
    test
  </span>
</div>
`;
